src/core/tests/type_prop/assign.cpp

```diff
@@ -78,3 +79,99 @@ TEST(type_prop, variable_comparison) {
     ASSERT_FALSE(variable1->get_info() == variable4->get_info());
     ASSERT_FALSE(variable1->get_info() == variable5->get_info());
 }
+
+TEST(type_prop, assign_v6_static_shape_match) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1, 2, 64, 64});
+    auto variable = std::make_shared<op::util::Variable>(
+        op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"});
+    std::shared_ptr<ov::op::v6::Assign> assign;
+    EXPECT_NO_THROW(assign = std::make_shared<ov::op::v6::Assign>(input, variable));
+
+    ASSERT_EQ(assign->get_element_type(), element::f32);
+    ASSERT_EQ(assign->get_shape(), (Shape{1, 2, 64, 64}));
+}
+
+TEST(type_prop, assign_v6_static_shapes_do_not_match) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, Shape{2, 2, 64, 64});
+    auto variable_info = op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+    std::shared_ptr<ov::op::v6::Assign> assign;
+    EXPECT_ANY_THROW(assign = std::make_shared<ov::op::v6::Assign>(input, variable));
+}
+
+TEST(type_prop, assign_v6_static_types_do_not_match) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::i32, Shape{1, 2, 64, 64});
+    auto variable_info = op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+    std::shared_ptr<ov::op::v6::Assign> assign;
+    EXPECT_ANY_THROW(assign = std::make_shared<ov::op::v6::Assign>(input, variable));
```

Comment:
```suggestion
    OV_EXPECT_THROW(std::ignore= std::make_shared<ov::op::v6::Assign>(input, variable), exp_exception, testing::HasSubstr(""));
```
As alternative this helper macro can be used from `test_asssertion.hpp` to check if expected exception has been thrown and also, optionally message of exception can be checked.